playbooks/roles/katello_provisioning/tasks/main.yml

```diff
@@ -1,18 +1,236 @@
+  register: katello_provisioning_centos7_json
+  ignore_errors: True
+
+- name: 'get CentOS 7 info'
   set_fact:
-    katello_provisioning_smart_proxy: "{{ katello_provisioning_smart_proxy_json.stdout|from_json }}"
+    katello_provisioning_centos7: "{{ katello_provisioning_centos7_json.stdout|from_json }}"
 
-# Setup Foreman with DNS, TFTP, DHCP
-- include: foreman.yml
-  when: katello_provisioning_foreman
+- name: 'update partition table for CentOS_Linux 7'
+  shell: >
+    {{ katello_provisioning_hammer }} os update --id {{ katello_provisioning_centos7.Id }} --partition-tables 'Kickstart default'
+  register: katello_provisioning_centos7_json
 
-# Setup Katello content - activation keys, repos, etc, and a host group
-- include: katello.yml
-  when: katello_provisioning_katello
+- name: 'find kickstart templates'
+  shell: >
+    {{ katello_provisioning_hammer }} --output=json template list
+    --search 'name ~ "Kickstart default"'
+  register: kickstart_templates_json
+
+- name: 'set kickstart templates'
+  set_fact:
+    kickstart_templates: "{{ kickstart_templates_json.stdout|from_json }}"
+
+- name: 'find katello kickstart templates'
+  shell: >
+    {{ katello_provisioning_hammer }} --output=json template list
+    --search 'name ~ "Katello Kickstart default"'
+  register: katello_kickstart_templates_json
+
+- name: 'set kickstart templates'
+  set_fact:
+    katello_kickstart_templates: "{{ katello_kickstart_templates_json.stdout|from_json }}"
+
+- name: 'associate kickstart templates to CentOS_Linux 7'
+  shell: >
+    {{ katello_provisioning_hammer }} template add-operatingsystem --id {{ item.Id }} --operatingsystem 'CentOS_Linux 7'
+  with_items: "{{ kickstart_templates }}"
+
+- name: 'set default templates for CentOS 7'
+  shell: >
+    {{ katello_provisioning_hammer }} os set-default-template --id {{ katello_provisioning_centos7.Id }} --config-template-id {{ item.Id }}
+  with_items: "{{ kickstart_templates }}"
+
+- name: 'overwrite default templates with katello specific ones for CentOS 7'
+  shell: >
+    {{ katello_provisioning_hammer }} os set-default-template --id {{ katello_provisioning_centos7.Id }} --config-template-id {{ item.Id }}
+  with_items: "{{ katello_kickstart_templates }}"
+
+# Host group
+- name: 'find hostgroup Forklift Katello CentOS 7'
+  shell: >
+    {{ katello_provisioning_hammer }} hostgroup info --name 'Forklift Katello CentOS 7'
+  register: katello_provisioning_hostgroup_katello_centos
+  ignore_errors: True
+
+- name: 'create Forklift Katello CentOS 7 host group'
+  shell: >
+    {{ katello_provisioning_hammer }} hostgroup create
+    --name "Forklift Katello CentOS 7"
+    --content-view "Default Organization View"
+    --lifecycle-environment Library
+    --query-organization "{{ katello_provisioning_organization }}"
+    --content-source-id {{ foreman_provisioning_smart_proxy.Id }}
+    --kickstart-repository-id 1
+    --operatingsystem "CentOS_Linux 7"
+    --medium 'CentOS mirror'
```

Comment:
Is this hostgroup supposed to use synced content? If so, this should be removed.